python/src/init/PyInit.cpp

```diff
@@ -14,23 +14,17 @@
  * limitations under the License.
  */
 
-#include "velox/py/lib/PyInit.h"
+#include "python/src/init/PyInit.h"
 #include "velox/common/memory/Memory.h"
 
 namespace facebook::velox::py {
 
-folly::once_flag initOnceFlag;
-
-void initializeVeloxMemoryOnce() {
-  // Enable full Velox stack trace when exceptions are thrown.
-  FLAGS_velox_exception_user_stacktrace_enabled = true;
-
-  velox::memory::initializeMemoryManager({});
-}
-
 void initializeVeloxMemory() {
-  // Initialize Velox once per process.
-  folly::call_once(initOnceFlag, initializeVeloxMemoryOnce);
+  if (not velox::memory::MemoryManager::testInstance()) {
+    // Enable full Velox stack trace when exceptions are thrown.
+    FLAGS_velox_exception_user_stacktrace_enabled = true;
+    velox::memory::initializeMemoryManager({});
```

Comment:
Shouldnt we add this under folly::call_once so that it isnt called multiple times ? Is library init guaranteed to be single threaded ?

Reply:
That was the previous solutio but that didn't work and threw runtime errors due to double init. I didn't investigate the reason as I don't think the minor overhead to `testInstance` is harming anything in the context of python. I would guess it's single threaded but I don't actually know.